Modules/_ssl.c

```diff
@@ -885,13 +891,15 @@ newPySSLSocket(PySSLContext *sslctx, PySocketSockObject *sock,
     self->server_hostname = NULL;
     self->err = err;
     self->exc = NULL;
+    self->tstate_mutex = (PyMutex){0};
 
     /* Make sure the SSL error state is initialized */
     ERR_clear_error();
 
-    PySSL_BEGIN_ALLOW_THREADS
+    Py_BEGIN_ALLOW_THREADS
```

Comment:
Use `PySSL_BEGIN_ALLOW_THREADS(ssl->ctx)` here.  The PySSLContext's ctx is being accessed by SSL_new so we should lock it.

Reply:
Would probably be good to add a test creating ssl contexts in multiple threads as well.